src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -312,12 +312,13 @@ public KruizeObject prepareRecommendations(int calCount) {
     public void generateRecommendations(KruizeObject kruizeObject) {
 
         for (K8sObject k8sObject : kruizeObject.getKubernetes_objects()) {
-            if (k8sObject.getExperimentType().equalsIgnoreCase(AnalyzerConstants.ExperimentTypes.NAMESPACE_EXPERIMENT)) {
+            String experimentType = kruizeObject.getExperimentType();
```

Comment:
I was thinking to have this function inside KruizeObject 
```
private boolean isContainerExperiment(String experimentType) {
    return experimentType == null || experimentType.equalsIgnoreCase(AnalyzerConstants.ExperimentTypes.CONTAINER_EXPERIMENT);
}

private boolean isNamespaceExperiment(String experimentType) {
    return experimentType != null && experimentType.equalsIgnoreCase(AnalyzerConstants.ExperimentTypes.NAMESPACE_EXPERIMENT);
}
```

and then use this function across everywhere like this
```

if (isContainerExperiment(experimentType)) {
    k8sObject = createContainerExperiment(kubernetesAPIObject); or
     generateRecommendationsBasedOnContainer(containerData, kruizeObject);
} else if (isNamespaceExperiment(experimentType)) {
    k8sObject = createNamespaceExperiment(kubernetesAPIObject);
    generateRecommendationsBasedOnNamespace
}
```

Reply:
Sorry for the misunderstanding. I've updated the code as per our discussion. Let me know if any further changes are needed.